src/data-structures/Hashmap.huff

```diff
@@ -62,6 +62,47 @@
     sha3                // [key]
 }
 
+/// @notice Calculate the slot from three keys
+#define macro GET_SLOT_FROM_KEYS_3D(mem_ptr) = takes(4) returns (1) {
+    // Input stack: [slot, key1, key2, key3]
+    // Load the data into memory
+    <mem_ptr>           // [<mem_ptr>, slot, key1, key2, key3]
+    mstore              // [key1, key2, key3]
+
+    // next byte
+    <mem_ptr> 0x20 add  // [<mem_ptr> + 32, key1, key2, key3]
```

Comment:
small opti but you could keep the value in the stack, as it is re-used two times after

Reply:
Not sure I understood that. How would the code look like then?

Reply:
You would dup `<mem_ptr> + 32` to be able to reuse it after with just a dup

Reply:
I believe I followed the same pattern as the other Hashmap.huff functions. Maintainers can chime in on whether that optimization is necessary/wanted.

Reply:
now there are three uses keeping it on the stack makes sense! With just two a dup1 and a swap are as expensive as another push (usually) so it is ignored.